### ISSUE TITLE
rdf_has/3 fails on string without explicit type

### ISSUE THREAD

**Issue by grsjst** [SWI-Prolog/packages-semweb#38]
test.ttl:
```
@prefix bastri: <https://bastri.inria.fr/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
<SR0192DR>
   bastri:siid "SR0192DR" ;
   bastri:n "1"^^xsd:integer.
```
session:
```
Welcome to SWI-Prolog (Multi-threaded, 64 bits, Version 7.3.22-29-gb68a39a)
Copyright (c) 1990-2015 University of Amsterdam, VU Amsterdam
SWI-Prolog comes with ABSOLUTELY NO WARRANTY. This is free software,
and you are welcome to redistribute it under certain conditions.
Please visit http://www.swi-prolog.org for details.

For help, use ?- help(Topic). or ?- apropos(Word).

?- use_module(library(semweb/rdf11)).
true.

?- use_module(library(semweb/turtle)).
true.

?- rdf_load('test.ttl').
% Parsed "test.ttl" in 0.00 sec; 2 triples
true.

?- rdf_has(S,'https://bastri.inria.fr/n',V^^T).
S = 'file:///home/geurts/tmp/dpei/db/SR0192DR',
V = 1,
T = 'http://www.w3.org/2001/XMLSchema#integer'.

?- rdf_has(S,'https://bastri.inria.fr/siid',V^^T).
false.
```

I expected the last query to succeed.  ...or am I missing something?

thanks,
Joost

**Comment by wouterbeek** [comment]
Hi Joost,

I cannot reproduce your issue because I cannot load your Turtle file.  According to Turtle syntax IRIs in prefix declarations must appear in between angular brackets.  Your file also seems to lack a subject term?

Maybe something went wrong while putting your data file in the Github comment box.  You can use three consecutive  backticks to start/end code blocks.

```
like this
```

**Comment by grsjst** [comment]
Hi Wouter,

Thanks for the tip! The brackets reappeared in all their glory (updated in the initial post).

**Comment by wouterbeek** [comment]
I can reproduce this one (also for `rdf/3`).  The mapping from/to `rdf_db` compound terms representing literals is quite thorny; I cannot see the fix for this.  Hopefully @JanWielemaker can take a look at this one.

@grsjst If you need to get around this issue now you can use the old `rdf_db` module.

**Comment by JanWielemaker** [comment]
Fixed with f2b4d3fb49eb87123e279fe2d214541858f8ea30.  Wouter is right about the code being _thorny_ ...
